core/src/main/java/org/jboss/qa/jcontainer/util/ProcessUtils.java

```diff
@@ -60,15 +60,22 @@ public static void killJavaByContainerId(long id) {
 	public static String getJavaPidByContainerId(long id) {
 		String result = null;
 		try {
+			final Long currentPid = PidUtils.getPID();
 			for (Map.Entry<String, String> entry : getJavaProcesses().entrySet()) {
 				final String pid = entry.getKey();
-				final Process p = Runtime.getRuntime().exec("jinfo -sysprops " + pid);
+				if (currentPid != null && String.valueOf(currentPid).equals(pid)) {
+					//it hangs jvm on Windows
+					log.debug("We are not going to execute jinfo for our pid " + currentPid);
+					continue;
+				}
+				final String command = "jinfo -sysprops " + pid;
+				log.info("Executing: " + command);
+				final Process p = Runtime.getRuntime().exec(command);
 				final BufferedReader is = new BufferedReader(new InputStreamReader(p.getInputStream()));
 				String line;
 				while ((line = is.readLine()) != null) {
-					if (line.contains(Container.JCONTAINER_ID) && line.contains(String.valueOf(id))) {
+					if (result == null && line.contains(Container.JCONTAINER_ID) && line.contains(String.valueOf(id))) {
 						result = pid;
-						break;
```

Comment:
we want to consume STDOUT of executed command. It could freeze if STDOUT buffer of executed process is overfull.